index.bs

```diff
@@ -58,5398 +68,5601 @@ urlPrefix: https://tc39.github.io/ecma262/; spec: ECMASCRIPT
+{{ReadableStreamBYOBReader/releaseLock()|byobReader.releaseLock()}}, or
+{{WritableStreamDefaultWriter/releaseLock()|writer.releaseLock()}} method, as appropriate.
+
+<h2 id="conventions">Conventions</h2>
 
-A <a>readable byte stream</a> has the ability to vend two types of readers: <dfn>default readers</dfn> and <dfn>BYOB
-readers</dfn>. BYOB ("bring your own buffer") readers allow reading into a developer-supplied buffer, thus minimizing
-copies. A non-byte readable stream can only vend default readers. Default readers are instances of the
-{{ReadableStreamDefaultReader}} class, while BYOB readers are instances of {{ReadableStreamBYOBReader}}.
+This specification depends on the Infra Standard. [[!INFRA]]
 
-Similarly, a <dfn lt="writer|writable stream writer">writable stream writer</dfn>, or simply writer, is an object that
-allows direct writing of <a>chunks</a> to a <a>writable stream</a>. Without a writer, a <a>producer</a> can only perform
-the high-level operations of <a lt="abort a writable stream">aborting</a> the stream or <a>piping</a> a readable stream
-to the writable stream. Writers are represented by the {{WritableStreamDefaultWriter}} class.
+This specification uses the [=abstract operation=] concept from [[!ECMASCRIPT]] for its internal
+algorithms. This includes treating their return values as [=completion records=], and the use of !
+and ? prefixes for unwrapping those completion records.
 
-<p class="note">Under the covers, these high-level operations actually use a reader or writer themselves.</p>
+This specification also uses the [=internal slot=] and [=record=] concepts and notation from
+ECMAScript. (Although, the internal slots are on Web IDL [=platform objects=] instead of ECMAScript
+objects.)
 
-A given readable or writable stream only has at most one reader or writer at a time. We say in this case the stream is
-<dfn lt="lock|locked to a reader|locked to a writer">locked</dfn>, and that the reader or writer is <dfn
-lt="active|active reader|active writer">active</dfn>. This state can be determined using the
-{{ReadableStream/locked|readableStream.locked}} or {{WritableStream/locked|writableStream.locked}} properties.
+<p class="note">The reasons for the usage of these foreign ECMAScript conventions are largely
+historical. We urge you to avoid following our example when writing your own web specifications.
 
-A reader or writer also has the capability to <dfn lt="release a lock|release a read lock|release a write lock">release
-its lock</dfn>, which makes it no longer active, and allows further readers or writers to be acquired. This is done via
-the {{ReadableStreamDefaultReader/releaseLock()|defaultReader.releaseLock()}},
-{{ReadableStreamBYOBReader/releaseLock()|byobReader.releaseLock()}}, or
-{{WritableStreamDefaultWriter/releaseLock()|writer.releaseLock()}} method, as appropriate.
+Finally, as in [[!ECMASCRIPT]], all numbers are represented as double-precision floating point
```

Comment:
Minor nitpick, but ECMAScript has three numeric types, including the spec-only "mathematical values". I'd explicitly reference the Number type, or go with IDL's `unrestricted double`.

Reply:
Thanks, I agree. I think this text originated from before the introduction of BigInt.